delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/routes/FilesRoutes.scala

```diff
@@ -0,0 +1,266 @@
+import monix.bio.IO
+import monix.execution.Scheduler
+
+/**
+  * The files routes
+  *
+  * @param identities    the identity module
+  * @param acls          the acls module
+  * @param organizations the organizations module
+  * @param projects      the projects module
+  * @param files         the files module
+  */
+final class FilesRoutes(
+    identities: Identities,
+    acls: Acls,
+    organizations: Organizations,
+    projects: Projects,
+    files: Files
+)(implicit
+    baseUri: BaseUri,
+    storageConfig: StorageTypeConfig,
+    s: Scheduler,
+    cr: RemoteContextResolution,
+    ordering: JsonKeyOrdering
+) extends AuthDirectives(identities, acls)
+    with CirceUnmarshalling {
+
+  import baseUri.prefixSegment
+
+  def routes: Route =
+    baseUriPrefix(baseUri.prefix) {
+      extractCaller { implicit caller =>
+        pathPrefix("files") {
+          concat(
+            // SSE files for all events
+            (pathPrefix("events") & pathEndOrSingleSlash) {
+              get {
+                operationName(s"$prefixSegment/files/events") {
+                  authorizeFor(AclAddress.Root, events.read).apply {
+                    lastEventId { offset =>
+                      emit(files.events(offset))
+                    }
+                  }
+                }
+              }
+            },
+            // SSE files for all events belonging to an organization
+            ((label | orgLabelFromUuidLookup(organizations)) & pathPrefix("events") & pathEndOrSingleSlash) { org =>
+              get {
+                operationName(s"$prefixSegment/files/{org}/events") {
+                  authorizeFor(AclAddress.Organization(org), events.read).apply {
+                    lastEventId { offset =>
+                      emit(files.events(org, offset).leftWiden[FileRejection])
+                    }
+                  }
+                }
+              }
+            },
+            (projectRef | projectRefFromUuidsLookup(projects)) { ref =>
+              concat(
+                // SSE files for all events belonging to a project
+                (pathPrefix("events") & pathEndOrSingleSlash) {
+                  get {
+                    operationName(s"$prefixSegment/files/{org}/{project}/events") {
+                      authorizeFor(AclAddress.Project(ref), events.read).apply {
+                        lastEventId { offset =>
+                          emit(files.events(ref, offset))
+                        }
+                      }
+                    }
+                  }
+                },
+                // Create a file without id segment
+                (post & pathEndOrSingleSlash & noParameter("rev") & parameter("storage".as[IdSegment].?)) { storage =>
+                  extractRequestEntity { entity =>
+                    operationName(s"$prefixSegment/files/{org}/{project}") {
+                      emit(Created, files.create(storage, ref, entity))
```

Comment:
In all the other routes, we do the authorization in the routes, but for files we do it in Files, it's a bit inconsistent.

Reply:
Yes it is. But there is a good reason for it. File authorization depends on storage and not on a hardcoded value. 

All the logic about fetching a storage, expanding its IdSegment, checking if it is deprecated, etc it is done inside the `Files` bundle. So is the authorization.

This leaves the routes less verbose, while the logic is concentrated in the bundle.